src/index.js

```diff
@@ -150,20 +152,23 @@ const makeDOMDriver =
       view$ => {
         validateDOMDriverInput(view$)
 
-        const rootElem$ =
-          most.create(
-            add =>
-              view$
-                .flatMap(parseTree)
-                .reduce(
-                  (prevView, newView) => {
-                    patch(prevView, newView)
-                    add(newView.elm)
-                    return newView
-                  }
-                  , rootElem
-                )
-          )
+        const rootElem$ = most.create(
+          (add, end, error) => {
+            view$
+              .map(parseTree)
+              .switch()
+              .reduce(
+                (prevView, newView) => {
+                  const vnode = patch(prevView, newView)
+                  add(vnode.elm)
+                  return vnode
+                },
+                rootElem
+              )
+              .then(end)
+              .catch(error)
+          }
+        )
```

Comment:
Technically no, but it got changed along the way as I was trying to figure out where some problems were coming from. At first I thought `add(newView.elm)` might be pushing the wrong Element. I also cleaned up based on things I've learned along the way in that process. 

I can revert these changes and submit it as a separate PR